api/v1alpha1/httproute_types.go

```diff
@@ -190,6 +195,19 @@ type HTTPRouteAction struct {
 	Extension *core.TypedLocalObjectReference `json:"extension"`
 }
 
+// HTTPReencrypt specifies parameters for re-encrypting the forwarded
+// connection.
+type HTTPReencrypt struct {
```

Comment:
This type would be an appropriate place to add a field to configure a client certificate for mTLS in the future.

Reply:
AppProtocol field is being added to k8s 1.18 and there might be a conflict for that here.
I think having this makes sense to me but we should have an explicit protocols field here to define what is the protocol to use when communicating with the upstream service. It can be added in the future. Just leaving a note here, not requesting any changes or advocating against this field.